upup/pkg/fi/cloudup/openstacktasks/lblistener.go

```diff
@@ -69,7 +69,6 @@ func NewLBListenerTaskFromCloud(cloud openstack.OpenstackCloud, lifecycle *fi.Li
 		}
 		listenerTask.Pool = poolTask
 		// TODO: Support Multiple?
-		break
```

Comment:
Wouldn't removing this change the `listenerTask.Pool` from being the first item in `lb.Pools` to the last item?

Reply:
Sorry, forget it.
NewLBListenerTaskFromCloud has not comment , so i has some questions about this function. 
This seems  that we need not to  use  for{}  . any advice?

Reply:
@hakman any advice?Thanks

Reply:
I would try this:
```diff
--- upup/pkg/fi/cloudup/openstacktasks/lblistener.go.old
+++ upup/pkg/fi/cloudup/openstacktasks/lblistener.go
@@ -66,10 +66,11 @@
 		poolTask, err := NewLBPoolTaskFromCloud(cloud, lifecycle, &pool, find.Pool)
 		if err != nil {
 			return nil, fmt.Errorf("NewLBListenerTaskFromCloud: Failed to create new LBListener task for pool %s: %v", pool.Name, err)
-		}
-		listenerTask.Pool = poolTask
-		// TODO: Support Multiple?
-		break
+		} else {
+			listenerTask.Pool = poolTask
+			// TODO: Support Multiple?
+			break
+		}
 	}
 	if find != nil {
 		// Update all search terms
```